### PR TITLE
[DO NOT MERGE] Ensure GOV.UK Design System has full test coverage prior to 1.6 release

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description

This ensures that the Boostrap and GOV.UK Design System implementation are fully tested. I've done this by doing the following: 

1. forced the `preview_design_system?` to always return true to force all tests to run in the design system
2. noted down out all tests that aren't in a legacy test file that failed
3. Either duplicated the test file and prepended one with legacy. Then gave the non-legacy files user the `Preview design system` permission or duplicated the test and gave the new tests user the permission
4. removed the forced permission

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: test/functional/admin/detailed_guides_controller_test/national_applicability_test.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class NationalApplicabilityTest < ActionController::TestCase
 
     setup do
       login_as create(:writer, organisation: create(:organisation))
+      current_user.permissions << "Preview design system"
```
Comment (ollietreend) [inline test/functional/admin/detailed_guides_controller_test/national_applicability_test.rb:9]: Would this work instead?

```suggestion
      login_as_preview_design_system_user :writer
```

Comment (ollietreend) [inline test/functional/admin/detailed_guides_controller_test/national_applicability_test.rb:9]: Would this work instead?

```suggestion
      login_as_preview_design_system_user :writer
```

**Review by ollietreend — APPROVED** [review]
This looks great. Really nice work spotting all the uncovered changes. And bonus points for getting through the gnarly National Inapplicabilities fixes! 💪🏻 🚀

File: test/functional/admin/detailed_guides_controller_test/national_applicability_test.rb (hunk 1 above)
Comment (ollietreend) [reply test/functional/admin/detailed_guides_controller_test/national_applicability_test.rb:9]: Ah, you might be right. I'd assumed `create(:writer)` would give the user an org automatically. But looking at [the factory](https://github.com/alphagov/whitehall/blob/ab98fb14d01fddd38abf79a430fb68dbcf365de8/test/factories/users.rb), I think I might be wrong on that!

No stress if it causes tests to fail – just leave it as is!

Comment (davidgisbey) [reply test/functional/admin/detailed_guides_controller_test/national_applicability_test.rb]: it went 🟢  so all good. Thanks for that

**Pull request merged**